conans/test/unittests/client/tools/scm/test_scm_base.py

```diff
@@ -53,4 +56,27 @@ def test_local_windows(self):
     def test_svn_ssh(self):
         self.assertEqual('svn+ssh://10.106.191.164/home/svn/shproject',
                          SCMBase._remove_credentials_url(
-                             'svn+ssh://username:password@10.106.191.164/home/svn/shproject'))
\ No newline at end of file
+                             'svn+ssh://username:password@10.106.191.164/home/svn/shproject'))
+
+
+class GetUrlWithCredentialsTest(unittest.TestCase):
+
+    def test_ssh(self):
+        scm = SCMBase()
+        self.assertEqual('ssh://github.com/conan-io/conan.git',
+                         scm.get_url_with_credentials("ssh://github.com/conan-io/conan.git"))
+
+    def test_ssh_username_password(self):
+        scm = SCMBase(username="dani", password="pass")
+        self.assertEqual('ssh://dani:pass@github.com/conan-io/conan.git',
+                         scm.get_url_with_credentials("ssh://github.com/conan-io/conan.git"))
+
+    def test_ssh_username(self):
+        scm = SCMBase(username="dani")
+        self.assertEqual('ssh://dani@github.com/conan-io/conan.git',
+                         scm.get_url_with_credentials("ssh://github.com/conan-io/conan.git"))
+
+    def test_ssh_password(self):
+        scm = SCMBase(password="pass")
```

Comment:
I am a bit confused a bit about this test. So adding the password, but not the user, will not add the password, but will not error either?

Reply:
This was the behavior before but I am not sure how the user could set only the password. Maybe providing the user already in the URL?

Reply:
Well, that is the point, the previous behavior was requiring both user+password. Now we changed it, and adding only the user is a new valid behavior, with a new test checking it. 

I'd say that we need to define a new behavior for only the password: either an error for this case, or check that there is a valid scenario, like the user is already in the URL, but that should probably be a new test too.